src/resolve/metadataResolver.ts

```diff
@@ -145,7 +147,19 @@ export class MetadataResolver {
       function: 'resolveComponent',
       path: fsPath,
     });
-    throw new SfError(messages.getMessage('error_could_not_infer_type', [fsPath]), 'TypeInferenceError');
+
+    // If a file ends with .xml and is not a metadata type, it is likely a package manifest
```

Comment:
you want this before the telemetry so we're not reporting lots of new package.xml resolution failures

Reply:
Good call